tests/fe/inf_fe_jacobi_test.C

```diff
@@ -0,0 +1,369 @@
+    std::unique_ptr<FEBase> inf_fe (FEBase::build_InfFE(dim, fe_type));
+    QGauss qrule (dim, fe_type.default_quadrature_order());
+    inf_fe->attach_quadrature_rule(&qrule);
+    const std::vector<std::vector<Real>> & phi = inf_fe->get_phi();
+    const std::vector<std::vector<RealGradient>> & dphi = inf_fe->get_dphi();
+
+    // Reinit on infinite elem
+    inf_fe->reinit(infinite_elem);
+
+    // Check phi values against reference values. The number of
+    // quadrature points and shape functions seem to both increase by
+    // 4 as the radial_order is increased.
+    //
+    // radial_order | n_qp | n_sf
+    // --------------------------
+    // FIRST        | 16   | 8
+    // SECOND       | 20   | 12
+    // THIRD        | 24   | 16
+    // FOURTH       | 28   | 20
+    // FIFTH        | 32   | 24
+
+    // Debugging print values so we can tabulate them
+    // auto n_qp = inf_fe->n_quadrature_points();
+    // auto n_sf = inf_fe->n_shape_functions();
+    // for (unsigned int i=0; i<n_sf; ++i)
+    //   for (unsigned qp=0; qp<n_qp; ++qp)
+    //     {
+    //       libMesh::out << "{" << i << "," << qp << "," << phi[i][qp] << "}," << std::endl;
+    //     }
+    // for (unsigned int i=0; i<n_sf; ++i)
+    //   for (unsigned qp=0; qp<n_qp; ++qp)
+    //     {
+    //       libMesh::out << "{" << i << "," << qp << ",Point("
+    //                    << dphi[i][qp](0)
+    //                    << ","
+    //                    << dphi[i][qp](1)
+    //                    << ","
+    //                    << dphi[i][qp](2)
+    //                    << ")},"
+    //                    << std::endl;
+    //     }
+
+    // Get reference vals for this Order/Family combination.
+    const TabulatedVals & val_table =
+      tabulated_vals[std::make_pair(radial_order, radial_family)];
+    const TabulatedGrads & grad_table =
+      tabulated_grads[std::make_pair(radial_order, radial_family)];
+
+    // Test whether the computed values match the tabulated values to
+    // the specified accuracy.
+    for (const auto & t : val_table)
+      CPPUNIT_ASSERT_DOUBLES_EQUAL(phi[t.i][t.qp], t.val, 1.e-4);
+    for (const auto & t : grad_table)
+      CPPUNIT_ASSERT_DOUBLES_EQUAL((dphi[t.i][t.qp] - t.grad).norm_sq(), 0., 1.e-4);
+
+    // Make sure there are actually reference values
+    if (val_table.empty())
+      libmesh_error_msg("No tabulated values found!");
+    if (grad_table.empty())
+      libmesh_error_msg("No tabulated gradients found!");
+
+#endif // LIBMESH_ENABLE_INFINITE_ELEMENTS
+  }
+
+  void setUp()
+  {
+    ////////////////////////////////////////////////////////////////////////////////
+    // Arbitrarily selected JACOBI_20_00 reference values.
+    ////////////////////////////////////////////////////////////////////////////////
+    tabulated_vals[std::make_pair(FIRST, JACOBI_20_00)] =
+      {
+        // These values are the same as Legendre
+        {0,9,0.0550016}, {1,5,0.41674},   {2,8,0.0147376}, {3,7,0.111665},
```

Comment:
Not that it's insufficient for this purpose, but why only 5 or 6 signficant decimals?

Reply:
Laziness? I just copy/pasted the values that were printed to `std::cout` by the test, so they were in that default format which seemed to have about 4 or 5 digits by default.